mods/ra/maps/soviet-08b/soviet08b-AI.lua

```diff
@@ -0,0 +1,176 @@
+	normal = { { "1tnk", "1tnk", "e3", "e3", "jeep"  }, { "2tnk", "e3", "e3", "jeep" } },
+	easy = { { "1tnk", "e1", "e1", "e3", "e3" }, { "e3", "e3", "jeep", "jeep" } }
+}
+
+WTransDelays =
+{
+	easy = 4,
+	normal = 2,
+	hard = 1
+}
+
+AttackGroup = { }
+AttackGroupSize = 10
+
+ProductionInterval =
+{
+	easy = DateTime.Seconds(30),
+	normal = DateTime.Seconds(20),
+	hard = DateTime.Seconds(10)
+}
+
+AlliedInfantry = { "e1", "e3" }
+AlliedVehiclesUpgradeDelay = DateTime.Minutes(15)
+AlliedVehicleType = "Normal"
+AlliedVehicles =
+{
+	Normal = { "jeep", "1tnk", "1tnk" },
+	Upgraded = { "2tnk", "arty" }
+}
+
+IdleHunt = function(unit) if not unit.IsDead then Trigger.OnIdle(unit, unit.Hunt) end end
+
+WTransWaves = function()
+	local way = Utils.Random(WTransWays)
+	local units = Utils.Random(WTransUnits)
+	local attackUnits = Reinforcements.ReinforceWithTransport(greece, "lst", units , way, { way[2], way[1] })[2]
+	Utils.Do(attackUnits, function(a)
+		Trigger.OnAddedToWorld(a, function()
+			a.AttackMove(SovietBase.Location)
+			IdleHunt(a)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(WTransDelays), WTransWaves)
+end
+
+SendAttackGroup = function()
+	if #AttackGroup < AttackGroupSize then
+		return
+	end
+
+	Utils.Do(AttackGroup, function(unit)
+		if not unit.IsDead then
+			Trigger.OnIdle(unit, unit.Hunt)
+		end
+	end)
+
+	AttackGroup = { }
+end
+
+ProduceInfantry = function()
+	if (GreeceTent1.IsDead or GreeceTent1.Owner ~= greece) and (GreeceTent2.IsDead or GreeceTent2.Owner ~= greece) then
+		return
+	end
+
+	greece.Build({ Utils.Random(AlliedInfantry) }, function(units)
+		table.insert(AttackGroup, units[1])
+		SendAttackGroup()
+		Trigger.AfterDelay(ProductionInterval[Map.LobbyOption("difficulty")], ProduceInfantry)
+	end)
+end
+
+ProduceVehicles = function()
+	if GreeceWarFactory.IsDead or GreeceWarFactory.Owner ~= greece then
+		return
+	end
+
+	greece.Build({ Utils.Random(AlliedVehicles[AlliedVehicleType]) }, function(units)
+		table.insert(AttackGroup, units[1])
+		SendAttackGroup()
+		Trigger.AfterDelay(ProductionInterval[Map.LobbyOption("difficulty")], ProduceVehicles)
+	end)
+end
+
+BringDDPatrol1 = function()
+	local units = Reinforcements.Reinforce(greece, DDPatrol1, ShipArrivePath, 0)
+	Utils.Do(units, function(unit)
+		Trigger.OnIdle(unit, function(patrols)
+			patrols.Patrol(DDPatrol1Path, true, 200)
+		end)
+	end)
+	if GreeceNavalYard.IsDead then
+		return
+	else
+		Trigger.OnAllKilled(units, function()
+			if Map.LobbyOption("difficulty") == "easy" then
+				Trigger.AfterDelay(DateTime.Minutes(7), BringDDPatrol1)
+			else
+				Trigger.AfterDelay(DateTime.Minutes(4), BringDDPatrol1)
+			end
+		end)
+	end
+end
+
+BringDDPatrol2 = function()
+	local units = Reinforcements.Reinforce(greece, DDPatrol2, ShipArrivePath, 0)
+	Utils.Do(units, function(unit)
+		Trigger.OnIdle(unit, function(patrols)
+			patrols.Patrol(DDPatrol2Path, true, 200)
+		end)
+	end)
+	if GreeceNavalYard.IsDead then
```

Comment:
Shouldn't that be checked inside the trigger? I assume the idea here was to not have additional reinforcements once the naval yard is destroyed.